MoltenVK/MoltenVK/Vulkan/vulkan.mm

```diff
@@ -2823,9 +2823,41 @@ MVK_PUBLIC_VULKAN_SYMBOL void vkDestroyPrivateDataSlot(
 	MVKTraceVulkanCallEnd();
 }
 
-MVK_PUBLIC_VULKAN_STUB(vkGetDeviceBufferMemoryRequirements, void, VkDevice, const VkDeviceBufferMemoryRequirements*, VkMemoryRequirements2*)
-MVK_PUBLIC_VULKAN_STUB(vkGetDeviceImageMemoryRequirements, void, VkDevice, const VkDeviceImageMemoryRequirements*, VkMemoryRequirements2*)
-MVK_PUBLIC_VULKAN_STUB(vkGetDeviceImageSparseMemoryRequirements, void, VkDevice, const VkDeviceImageMemoryRequirements*, uint32_t*, VkSparseImageMemoryRequirements2*)
+MVK_PUBLIC_VULKAN_SYMBOL void vkGetDeviceBufferMemoryRequirements(
+	VkDevice                                    device,
+	const VkDeviceBufferMemoryRequirements*     pInfo,
+	VkMemoryRequirements2*                      pMemoryRequirements) {
+	
+	MVKTraceVulkanCallStart();
+	auto* mvkDev = MVKDevice::getMVKDevice(device);
+	MVKBuffer mvkBuff(mvkDev, pInfo->pCreateInfo);
+	mvkBuff.getMemoryRequirements(pMemoryRequirements);
```

Comment:
I'd prefer to have a method on `MVKDevice` that fills out the requirements, instead of having to create a temporary resource and use that--which is exactly what this functionality is supposed to enable avoiding in the first place. But, I suppose that could wait for a future refactoring.

Reply:
I hear you, so I dug into it a bit.

Unfortunately, this looks like it would be a particularly messy refactoring, especially for `MVKImage` and its planes.

And AFAICT, through the magic of lazy init, we don't create any Metal resources during `MVKImage` or `MVKBuffer` construction. So the cost is mostly limited to `pCreateInfo` processing. For `MVKBuffer`, that cost is particularly trivial. For `MVKImage`, that processing is fairly complicated, but it's not clear how much could be skipped by refactoring.

Previously, I did modify the original code to instantiate `MVKBuffer` & `MVKImage` inline, in an attempt to reduce allocs.

So, while I agree in theory, AFAICT this would be a bunch of messy work for limited benefit, on a maintenance extension. 😉